src/RuleSet/DeclarationBlock.php

```diff
@@ -44,14 +44,21 @@ public static function parse(ParserState $parserState, ?CSSList $list = null): ?
             do {
                 $selectorParts[] = $parserState->consume(1)
                     . $parserState->consumeUntil(['{', '}', '\'', '"'], false, false, $comments);
-                if (\in_array($parserState->peek(), ['\'', '"'], true) && \substr(\end($selectorParts), -1) != '\\') {
-                    if (!isset($stringWrapperCharacter)) {
-                        $stringWrapperCharacter = $parserState->peek();
-                    } elseif ($stringWrapperCharacter === $parserState->peek()) {
-                        unset($stringWrapperCharacter);
-                    }
+                $nextCharacter = $parserState->peek();
+                switch ($nextCharacter) {
+                    case '\'':
+                        // The fallthrough is intentional.
+                    case '"':
+                        if (!\is_string($stringWrapperCharacter)) {
```

Comment:
Ah, we need to initialize `$stringWrapperCharacter` to `null` before the `do` to make this work.